lib/elixir/lib/io/ansi/docs.ex

```diff
@@ -53,6 +55,37 @@ defmodule IO.ANSI.Docs do
     newline_after_block()
   end
 
+  @doc """
+  Prints documentation metadata (only `since` and `deprecated` for now).
+
+  See `default_options/0` for docs on the supported options.
+  """
+  @spec print_metadata(map, keyword) :: :ok
+  def print_metadata(metadata, options \\ []) when is_map(metadata) do
+    options = Keyword.merge(default_options(), options)
+    print_each_metadata(metadata, options) && newline_after_block()
+  end
+
+  defp print_each_metadata(metadata, options) do
+    Enum.reduce(metadata, false, &print_single_metadata({&1, options}, &2))
+  end
+
+  defp print_single_metadata({{:deprecated, value}, options}, _printed) do
+    [metadata_label(:deprecated, options) | String.split(value, @spaces)]
+    |> write_with_wrap(options[:width], "            ", true)
+  end
+
+  defp print_single_metadata({{:since, value}, options}, _printed) do
+    IO.puts([metadata_label(:since, options), " ", value])
+  end
+
+  # only printing deprecated and since for now
+  defp print_single_metadata(_, printed), do: printed
+
+  defp metadata_label(key, options) do
+    "#{color(:doc_metadata, options)}#{key}:#{IO.ANSI.reset()}"
```

Comment:
What do you think about? `color(:doc_metadata, options) <> "#{key}:" <> IO.ANSI.reset()`

Reply:
@fertapric are you recommending this because you think it looks clearer, or for performance benefits?

Reply:
Because I think it looks clearer :)

Reply:
I agree, changing it. :)

Reply:
On second thought, we are constructing a string out of 4 elements, so I think the use of interpolation is ok. Changing this would require squash or rebase to make the history cleaner, and a new CI run, etc. While those are not significant overheads, in this case the benefit is not significant either to warrant them. I'll change this as well if other things come up too. Otherwise leaving this for another time. :)

Talking about overheads while writing an essay as a reply to a trivial change request. :)

Reply:
Ok, no problem.

> Changing this would require squash or rebase to make the history cleaner

I think José squashes everything into a single commit when he merges the PR (unless you are talking about the history for reviewing the PR)

Reply:
Not always. If the history in the PR is already clean he might not do it. :)